### PR TITLE
Optional BIP91/SEGSIGNAL support (disabled by default)

### PR DESCRIPTION
**Pull request by luke-jr**
This should probably only be used by miners.

### CODE REVIEW

**Comment by mkwia** [comment]
Concept ACK.

File: qa/rpc-tests/bip91.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        assert_equal(self.get_bip9_status('segsignal')['status'], 'started')
+        assert_equal(self.get_bip9_status('segsignal')['since'], 48)
+        assert_equal(self.get_bip9_status('segwit')['status'], 'started')
+        assert_equal(self.get_bip9_status('segwit')['since'], 144)
+        tmpl = self.nodes[0].getblocktemplate({})
+        assert_equal(tmpl['version'], 0x10000001|0x20000012)
+
+
+        # Test 3
+        # 28 out of 48 signal bit 4 to achieve LOCKED_IN
```
Comment (dougyoung) [inline qa/rpc-tests/bip91.py:77]: Should be `29 out of 48 ...`

Comment (luke-jr) [reply qa/rpc-tests/bip91.py:77]: @jl2012

Comment (jl2012) [reply qa/rpc-tests/bip91.py:77]: fixed here: https://github.com/jl2012/bitcoin/commit/15f38f574faf74f5daafbb81148c3396918cd563

**Comment by hsjoberg** [comment]
Concept ACK.

In regards to recent developments, I guess it makes sense to help the btc1 project properly propagate their blocks to prevent network fragmentation and consensus failures.

**Comment by kek-coin** [comment]
Should perform preferential peering with other BIP91 nodes.

**Comment by hsjoberg** [comment]
Service bit 7 (1 << 7) is currently being used for btc1 full nodes.
https://github.com/btc1/bitcoin/pull/87/files#diff-1df47808b82268c616535e7b20216dfbR272

**Pull request closed without merging**